### PR TITLE
chore: replace prettier with dprint

### PR DESCRIPTION
**Pull request by johnstcn**
Trying out https://github.com/dprint/dprint for formatting instead of prettier.
Also threw in `shfmt` for good measure.

Note:
* I had to rename `Dockerfile.$TAG -> $TAG.Dockerfile` due to an existing issue with [dprint-plugin-dockerfile](https://github.com/dprint/dprint-plugin-dockerfile/issues/6)

### CODE REVIEW

File: scripts/check_unstaged.sh
Comment (johnstcn) [inline scripts/check_unstaged.sh:1]: not technically needed, can remove

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -49,16 +49,8 @@ jobs:
...
-        uses: actions/setup-node@v4
-        with:
-          node-version: 20.x
-
-      - name: Install Node.js modules
-        run: yarn install
-
       - name: Check formatting
-        run: yarn format:check
+        uses: dprint/check@v2.2
```
Comment (johnstcn) [inline .github/workflows/build.yaml:53]: https://github.com/marketplace/actions/dprint-check-action

File: dprint.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "markdown": {
+  },
+  "dockerfile": {
+  },
+  "excludes": [
+    "deprecated/*"
+  ],
+  "plugins": [
+    "https://plugins.dprint.dev/markdown-0.17.0.wasm",
```
Comment (matifali) [inline dprint.json:10]: How do we keep these updated? O don't think dependable support this.

File: scripts/check_unstaged.sh
Comment (matifali) [reply scripts/check_unstaged.sh:1]: Let's remove it then.

**Review by matifali — APPROVED** [review]
That's awesome 😎.

File: dprint.json (hunk 2 above)
Comment (johnstcn) [reply dprint.json:10]: Currently there isn't any automated way, but apparently warning about outdated plugin versions is on the roadmap.

**Comment by matifali** [comment]
Is packages.json still required as there is no node anymore?

**Comment by johnstcn** [comment]
> Is packages.json still required as there is no node anymore?

missed removing that, but it can also go

**Pull request merged**